packages/ckeditor5-ui/src/focuscycler.ts

```diff
@@ -274,7 +280,8 @@ export default class FocusCycler extends EmitterMixin() {
 	 * @returns
 	 */
 	private _focus( view: FocusableView | null, direction: 1 | -1 ) {
-		if ( view ) {
+		// TBD
+		if ( view && this.focusTracker.focusedElement !== view.element ) {
```

Comment:
Yup, in particular it is used in the following scenario:

1.  There's only one focusable item that is already focused.
2.  `focusNext()` is executed.
3.  It recognizes there's nothing else to focus, so it returns (see [https://github.com/ckeditor/ckeditor5/pull/15285#discussion_r1395747487).](https://github.com/ckeditor/ckeditor5/pull/15285#discussion_r1395747487).) But it triggers the `forwardCycle` event.
4.  On `forwardCycle` event, `focusCycler` executes `focusFirst()`, which in turn executes `_focus()`. 
5.  So even though we cut the rope in one place, the element is still about to be focused. It would break the scenario discussed in the mentioned comment (when someone may want to move the focus to the outer cycler).

Unit test: `'does not refocus if there is only one focusable item'`.

Reply:
Would you add a short inline comment about that?